nominal/nominal.py

```diff
@@ -455,12 +455,36 @@ def __init__(
             else:
                 self.datasets = datasets
 
-        mins = []
-        maxs = []
-        for ds in self.datasets.values():
-            mins.append(ds["_python_datetime"].min())
-            maxs.append(ds["_python_datetime"].max())
-        self.datasets_domain = dict(START=min(mins), END=max(maxs))
+        start_dt = datetime.min
+        end_dt = datetime.min
+
+        # if you pass in a start, we use that
+        # if you don't, we can compute the start from the datasets,
+        # but only if they all use absolute timestamps
+        if start is None:
+            if any(ds.relative for ds in self.datasets.values()):
+                raise ValueError("runs must provide a start time if any dataset uses relative timestamps")
+            start_dt = min(ds["_python_datetime"].min() for ds in self.datasets.values())
+        else:
+            start_dt = parser.parse(start)
+
+        # if you pass in an end, we use that
+        # if you don't, we can compute the end from the datasets
+        # if they use relative timestamps, we can compute the end since we know the start
+        if end is None:
+            for ds in self.datasets.values():
+                if ds.relative:
+                    assert ds.ts_col in ds.columns
+                    assert ds.relative_units == "SECONDS", "relative timestamps are only currently supported in seconds"
+                    absolute_end = start_dt + timedelta(seconds=ds[ds.ts_col].max())
+                    end_dt = max(end_dt, absolute_end)
+                else:
+                    absolute_end = ds["_python_datetime"].max()
+                    end_dt = max(end_dt, absolute_end)
```

Comment:
absolute_end here may be tz-aware, and for the first iteration end_dt is tz-naive, and thus cannot be compared.

we can do the naive conversion, but unclear if users expect the timestamps to be local or UTC